pkg/cluster/initialize.go

```diff
@@ -41,13 +44,24 @@ func (u *simpleUpgrader) writeBlob(blobName string, cs *api.OpenShiftManagedClus
 	bsc := u.storageClient.GetBlobService()
 	c := bsc.GetContainerReference(ConfigContainerName)
 	b := c.GetBlobReference(blobName)
-
 	json, err := json.Marshal(cs)
 	if err != nil {
 		return err
 	}
-
-	return b.CreateBlockBlobFromReader(bytes.NewReader(json), nil)
+	ctx := context.Background()
+	return wait.PollImmediateUntil(1*time.Second,
+		func() (bool, error) {
+			err = b.CreateBlockBlobFromReader(bytes.NewReader(json), nil)
```

Comment:
can the retry somehow go into pkg/azureclient/storage so that the caller doesn't remember to have to do this?  this is how we handle it in the pkg/azureclient and in the kubeclient?

in the general case you'll need to do some magic so that you don't lose the contents of the reader if you get a network error half way through transmission (ugh)